include/boost/date_time/gregorian/greg_date.hpp

```diff
@@ -120,7 +120,7 @@ namespace gregorian {
     date end_of_month() const
     {
       ymd_type ymd = year_month_day();
-      short eom_day =  gregorian_calendar::end_of_month_day(ymd.year, ymd.month);
+      unsigned short eom_day =  gregorian_calendar::end_of_month_day(ymd.year, ymd.month);
```

Comment:
looks right to me. CI  check issue doesn't seem related...

Reply:
going to be off grid for a few days, will approve after or James can if he thinks its right.